### PR TITLE
[Merged by Bors] - refactor: use consistent names for forgetful smul structures

### PR DESCRIPTION
**Pull request by eric-wieser**
Moves:
- `AddCommMonoid.natModule` -> `AddCommMonoid.toNatModule`
- `AddCommGroup.intModule` -> `AddCommGroup.toIntModule`
- `algebraNat` -> `Semiring.toNatAlgebra`
- `algebraInt` -> `Ring.toIntAlgebra`
- `algebraRat` -> `DivisionRing.toRatAlgebra`
- `AddCommGroup.natModule.unique` -> `AddCommMonoid.uniqueNatModule`
- `AddCommGroup.intModule.unique` -> `AddCommGroup.uniqueIntModule`

The premise here is that the conversion from `AddCommGroup` to `AddCommMonoid` is not really any different from converting from `AddCommGroup` to `Module ℤ`; and so both should be named with the `to*` naming convention.

Tihs also brings the naming of the `Module` and `Algebra` instances into alignment.

I chose `toNatModule` instead of `toModuleNat` to match the prose, ℕ-module.
This fits a common naming convention if you view `Module` as infix, as we do with various other lean functions without actual infix notation.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Algebra/Basic.lean [hunk 1]
```diff
@@ -241,10 +241,10 @@ section Int
 variable (R : Type*) [Ring R]
 
 -- Lower the priority so that `Algebra.id` is picked most of the time when working with
--- `ℤ`-algebras. This is only an issue since `Algebra.id ℤ` and `algebraInt ℤ` are not yet defeq.
--- TODO: fix this by adding an `ofInt` field to rings.
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Basic.lean:245]: This was fixed long ago :)

**Comment by YaelDillies** [comment]
Should `toNatModule` be `toModuleNat`, while we are at it?

**Comment by eric-wieser** [comment]
I'd argue no, because `NatModule` is much closed to the prose, "$$\mathbb{N}$$-module".

**Comment by eric-wieser** [comment]
(edited the description with that justification)

**Comment by YaelDillies** [comment]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**